ambassador/src/main/java/com/iluwatar/ambassador/ServiceAmbassador.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * The MIT License
+ * Copyright (c) 2014-2016 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.ambassador;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.Thread.sleep;
+
+/**
+ *
+ * ServiceAmbassador provides an interface for a ({@link Client}) to access ({@link RemoteService}).
+ * The interface adds logging, latency testing and usage of the service in a safe way that will not
+ * add stress to the remote service when connectivity issues occur.
+ *
+ */
+public class ServiceAmbassador implements RemoteServiceInterface {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServiceAmbassador.class);
+  private static final int RETRIES = 3;
+  private static final int DELAY_MS = 3000;
+
+  ServiceAmbassador() {
+
+  }
+
+  @Override
+  public long doRemoteFunction(int value) {
+
+    return safeCall(value);
+  }
+
+  private long checkLatency(int value) {
+    RemoteService service = RemoteService.getRemoteService();
+    long startTime = System.currentTimeMillis();
+    long result = service.doRemoteFunction(value);
```

Comment:
Could be written on one line `RemoteService.getRemoteService().doRemoteFunction(value)`